cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -0,0 +1,621 @@
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseBadRequest, HttpResponseNotFound
+from django.shortcuts import redirect
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_http_methods, require_POST
+from opaque_keys.edx.keys import AssetKey, CourseKey
+from pymongo import ASCENDING, DESCENDING
+
+from common.djangoapps.edxmako.shortcuts import render_to_response
+from common.djangoapps.student.auth import has_course_author_access
+from common.djangoapps.util.date_utils import get_default_time_display
+from common.djangoapps.util.json_request import JsonResponse
+from openedx.core.djangoapps.contentserver.caching import del_cached_content
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from xmodule.contentstore.content import StaticContent  # lint-amnesty, pylint: disable=wrong-import-order
+from xmodule.contentstore.django import contentstore  # lint-amnesty, pylint: disable=wrong-import-order
+from xmodule.exceptions import NotFoundError  # lint-amnesty, pylint: disable=wrong-import-order
+from xmodule.modulestore.django import modulestore  # lint-amnesty, pylint: disable=wrong-import-order
+from xmodule.modulestore.exceptions import ItemNotFoundError  # lint-amnesty, pylint: disable=wrong-import-order
+
+from .exceptions import AssetNotFoundException, AssetSizeTooLargeException
+from .utils import reverse_course_url, get_files_uploads_url
+from .toggles import use_new_files_uploads_page
+
+
+REQUEST_DEFAULTS = {
+    'page': 0,
+    'page_size': 50,
+    'sort': 'date_added',
+    'direction': '',
+    'asset_type': '',
+    'text_search': '',
+}
+
+
+def handle_assets(request, course_key_string=None, asset_key_string=None):
+    '''
+    The restful handler for assets.
+    It allows retrieval of all the assets (as an HTML page), as well as uploading new assets,
+    deleting assets, and changing the 'locked' state of an asset.
+
+    GET
+        html: return an html page which will show all course assets. Note that only the asset container
+            is returned and that the actual assets are filled in with a client-side request.
+        json: returns a page of assets. The following parameters are supported:
+            page: the desired page of results (defaults to 0)
+            page_size: the number of items per page (defaults to 50)
+            sort: the asset field to sort by (defaults to 'date_added')
+            direction: the sort direction (defaults to 'descending')
+            asset_type: the file type to filter items to (defaults to All)
+            text_search: string to filter results by file name (defaults to '')
+    POST
+        json: create (or update?) an asset. The only updating that can be done is changing the lock state.
```

Comment:
> (or update?)

I know you didn't write this docstring, but this question mark is driving me crazy. Are we uncertain whether POST can update an asset? Is it possible, but undesirable behavior? If you know the answer, please clarify this line.